src/drivers/MongoDriver.ts

```diff
@@ -2473,11 +2470,68 @@ export class MongoDriver implements DataStore {
       children,
       revision: record.revision,
     });
+    return learningObject;
+  }
 
+  /**
+   * Generates released Learning Object from Document
+   *
+   * @private
+   * @param {User} author
+   * @param {ReleasedLearningObjectDocument} record
+   * @param {boolean} [full]
+   * @returns {Promise<LearningObject>}
+   * @memberof MongoDriver
+   */
+  private async generateReleasedLearningObject(
+    author: User,
+    record: LearningObjectDocument,
+    full?: boolean,
+  ): Promise<LearningObject> {
+    // Logic for loading any learning object
+    let learningObject: LearningObject;
+    let materials: LearningObject.Material;
+    let contributors: User[] = [];
+    let children: LearningObject[] = [];
+    // Load Contributors
+    if (record.contributors && record.contributors.length) {
+      contributors = await Promise.all(
+        record.contributors.map(userId => this.fetchUser(userId)),
+      );
+    }
+    // If full object requested, load up non-summary properties
+    if (full) {
+      // Logic for loading 'full' learning objects
+      materials = <LearningObject.Material>record.materials;
+      for (let i = 0; i < record['outcomes'].length; i++) {
```

Comment:
The `ReleasedLearningObjectDocument` type should have outcomes, can we use dot notation here instead?
```suggestion
      for (let i = 0; i < record.outcomes.length; i++) {
```